src/FileInput/SelectedFile.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import propTypes from 'prop-types'
+
+import { colors, spacers } from '../theme.js'
+import { AttachFile } from '../icons/AttachFile.js'
+import { Loading } from '../icons/Status.js'
+
+const SelectedFile = ({
+    label,
+    onRemove,
+    removeText,
+    loading,
+    onCancel,
+    cancelText,
+}) => (
+    <p>
+        <span className="icon">{loading ? <Loading /> : <AttachFile />}</span>
+
+        <span className="text">
+            <span className="label">{label}</span>
+
+            {loading && onCancel && cancelText && (
+                <span className="action" onClick={onCancel}>
+                    {cancelText}
+                </span>
+            )}
+
+            {!loading && (
+                <span className="action" onClick={onRemove}>
+                    {removeText}
+                </span>
+            )}
+        </span>
+
+        <style jsx>{`
+            p {
+                display: flex;
+                margin: 0;
+                padding-top: ${spacers.dp8};
+                line-height: 14px;
+            }
+            .icon > :global(svg) {
+                width: 12px;
+                height: 12px;
+                line-height: 12px;
+                margin-right: 4px;
+                flex-grow: 0;
+                flex-shrink: 0;
+            }
+            .text {
+                flex-grow: 1;
+                flex-shrink: 1;
+            }
+            .label {
+                font-size: 14px;
+                color: ${colors.grey900};
+                margin-right: 12px;
```

Comment:
I think this should be `margin-left` on the `.action`.
It's causing a margin that's outside of the `FileInput` box when the file name is displayed in multiple line.
This might cause that the `removeText` will have an oddly looking margin when it falls into its own line,
but I think an internal layout issue is less of a problem than one that can influence the elements in the document flow.

Reply:
> This might cause that the removeText will have an oddly looking margin when it falls into its own line
Yes that's why I changed this from a margin-left on the remove/cancel-link to a margin-right on the text.

I see the problem you are describing with the margin when I inspect the DOM... But if you inspect the parent elements, it's all correct. I don't think this is an actual problem. If it would be, we'd already be seeing it in the "File list" story.

I don't like having a margin-left on the link, for the reasons you mentioned, but I think I can fix it in another way, using the ::after pseudo element. That way we'll both be happy.